cmd/smd/smd-api.go

```
@@ -2495,12 +2538,369 @@ func (s *SmD) doRedfishEndpointsPost(w http.ResponseWriter, r *http.Request) {
+					},
+					RfEndpointFQDN:        "",
+					URL:                   system.URI,
+					ComponentEndpointType: "ComponentEndpointComputerSystem",
+					Enabled:               enabled,
+					RedfishSystemInfo: &rf.ComponentSystemInfo{
+						Actions:    nil,
+						EthNICInfo: addEthernetInterfacesToNICInfo(system.EthernetInterfaces, enabled),
+					},
+				}
+			)
+
+			// add the corresponding CompEthInterfaceV2 for each ComponentEndpoint created
+			createCompEthInterfacesV2(component, system.EthernetInterfaces)
+			knownCEs[system.UUID] = componentEndpoint.ComponentDescription.ID
+			ceNum++
+
+			// components
+			rowsAffected, err := s.db.InsertComponent(&component)
+			if err != nil {
+				sendJsonError(w, http.StatusInternalServerError,
+					fmt.Sprintf("failed to insert %d component(s): %v", rowsAffected, err))
+
+				// upsert here to keep allow returning error for duplicates when not forcing updates
+				_, err := s.db.UpsertComponents([]*base.Component{&component}, false)
+				if err != nil {
+					return fmt.Errorf("failed to update component: %w", err)
+				}
+				return fmt.Errorf("failed to insert %d component(s): %v", rowsAffected, err)
+			}
+
+			// component endpoints
+			err = s.db.UpsertCompEndpoint(&componentEndpoint)
+			if err != nil {
+				sendJsonError(w, http.StatusInternalServerError,
+					fmt.Sprintf("failed to upsert component endpoint: %v", err))
+				return fmt.Errorf("failed to upsert component endpoint: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// getSchemaVersion() tries to extract the schema version from the JSON data.
+// Used for ochami where discovery is diabled
+func (s *SmD) getSchemaVersion(w http.ResponseWriter, data []byte) int {
+	var (
+		schemaVersion int = 0 // default to 0
+		root          map[string]any
+		ok            bool
+		err           error
+	)
+
+	// unmarshal JSON to root
+	err = json.Unmarshal(data, &root)
+	if err != nil {
+		sendJsonError(w, http.StatusInternalServerError,
```

Return error instead of trying to continue.